src/reporting/reportingToken.se

```diff
@@ -81,8 +80,26 @@ def migrateLosingTokens():
     require(self.getBranch().getForkingMarket() != self.market)
     require(self.market.getFinalWinningReportingToken() != self)
     self.supply = 0
-    reputationSupply = self.getReputationToken().balanceOf(self)
-    self.getReputationToken().transfer(self.market.getFinalWinningReportingToken(), reputationSupply)
+    # Send losing reporting token REP to correct dispute bonds until bond holders are paid out (either all REP are sent or they hit 2x limit)
+    # Pay out to earlier disputers as much as possible before paying out to later disputers
+    reputationToken = self.getReputationToken()
+    automatedReporterDisputeBondToken = self.market.getAutomatedReporterDisputeBondToken()
+    limitedReportersDisputeBondToken = self.market.getLimitedReportersDisputeBondToken()
+    allReportersDisputeBondToken = self.market.getAllReportersDisputeBondToken()
+    if (automatedReporterDisputeBondToken and automatedReporterDisputeBondToken.getDisputedPayoutDistributionHash() != self.market.getFinalPayoutDistributionHash()):
+        amountNeeded = automatedReporterDisputeBondToken.getBondRemainingToBePaidOut() - reputationToken.balanceOf(automatedReporterDisputeBondToken)
+        amountToTransfer = min(amountNeeded, reputationToken.balanceOf(self))
+        if (amountToTransfer > 0):
+            reputationToken.transfer(automatedReporterDisputeBondToken, amountToTransfer)
+            log(type=Transfer, self, automatedReporterDisputeBondToken, amountToTransfer)
+    if (limitedReportersDisputeBondToken and limitedReportersDisputeBondToken.getDisputedPayoutDistributionHash() != self.market.getFinalPayoutDistributionHash()):
+        amountNeeded = limitedReportersDisputeBondToken.getBondRemainingToBePaidOut() - reputationToken.balanceOf(limitedReportersDisputeBondToken)
+        amountToTransfer = min(amountNeeded, reputationToken.balanceOf(self))
+        if (amountToTransfer > 0):
+            reputationToken.transfer(limitedReportersDisputeBondToken, amountToTransfer)
+            log(type=Transfer, self, limitedReportersDisputeBondToken, amountToTransfer)
+    if (reputationToken.balanceOf(self) > 0):
+        reputationToken.transfer(self.market.getFinalWinningReportingToken(), reputationToken.balanceOf(self))
```

Comment:
Consider extracting out an `amountToTransfer` variable like in the above two blocks so you don't have to call `balanceOf` twice back-to-back.